include/seqan3/range/decorator/gap_decorator_anchor_set.hpp

```diff
@@ -0,0 +1,525 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+ /*!\file
+  * \author Marie Hoffmann <marie.hoffmann AT fu-berlin.de>
+  * \brief Contains gap decorator to annotate sequences with gaps using a set.
+  */
+
+#pragma once
+
+#include <set>
+#include <tuple>
+#include <type_traits>
+
+#include <range/v3/all.hpp>
+
+#include <seqan3/alignment/aligned_sequence/aligned_sequence_concept.hpp>
+#include <seqan3/alphabet/concept.hpp>
+#include <seqan3/alphabet/gap/gapped.hpp>
+#include <seqan3/range/container/concept.hpp>
+#include <seqan3/range/detail/random_access_iterator.hpp>
+
+namespace seqan3
+{
+
+/*!\brief A gap decorator allows the annotation of sequences with gap symbols
+ * leaving the underlying sequence unmodified.
+ * \tparam inner_type The type of alphabet sequences that will be stored.
+ * \implements seqan3::aligned_sequence_concept
+ * \ingroup decorator
+ *
+ * This class may be used whenever you want to store or compute an alignment.
+ * The underlying (ungapped sequence) remains unmodified, and is augmented by
+ * gap information. Iterating over a gap_decorator sequences behaves as if you
+ * had a rolled out, aligned sequence with alphabet and gap symbols.
+ *
+ * \details
```

Comment:
Could you use latex here for the O annotations and so on?
e.g.
https://github.com/seqan/seqan3/blob/8df4ff7e5ce54aeaa8604db623e621b144180829/include/seqan3/search/fm_index/fm_index.hpp#L73